polygon/sync/execution_client.go

```diff
@@ -44,11 +46,16 @@ type ExecutionClient interface {
 }
 
 type executionClient struct {
-	client executionproto.ExecutionClient
+	client      executionproto.ExecutionClient
+	blockReader *freezeblocks.BlockReader
```

Comment:
This is breaking the abstraction. I would prefer if we don't do this. I would like to avoid to have `BlockReader` dependency in the Astrid packages.

Reply:
Instead my idea/vision about this was to go via the ExecutionEngine.Ready API